### PR TITLE
@alloy => Make sure not to fetch follow suggestions unless specified

### PR DESCRIPTION
**Pull request by mzikherman**
Fixes https://artsyproduct.atlassian.net/secure/RapidBoard.jspa?rapidView=57&projectKey=DISCO&view=planning&selectedIssue=DISCO-531

There was already a prop being used in the component to control rendering the modal w/ suggestions, this adds a GraphQL variable to suppress fetching that.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit db0960f181872e21f03759438c6436f279c675bd

https://deploy-preview-1627--artsy-reaction.netlify.com

**Comment by mzikherman** [comment]
The static nature of the compiled queries is great here. You can see with all the removed lines, that a bunch of places that query for Follow stuff no longer include the suggestions schema, with the artwork app still including it.

**Comment by alloy** [comment]
> The static nature of the compiled queries is great here. You can see with all the removed lines, that a bunch of places that query for Follow stuff no longer include the suggestions schema, with the artwork app still including it.

Agreed! And looking through those I started to wonder if this can’t be simplified. Correct me if I’m wrong, but shouldn’t it really be included in all the ‘sidebar’ related queries? (And are those in fact not simply queries only due to work not being finished yet on the artwork page, in the end those would go away and all be queried through the `ArtworkApp`?)

If so, I think there’s no need to define the variable at the operation level and pass it all the way down, but only override the default `false` value in the artwork sidebar fragment. I’ll push a PR to illustrate this.

**Comment by alloy** [comment]
After making a branch and pushing it I realised that your branch exists on a fork, which means I would have to make a fork myself, push there, struggle with the GH UI to open a PR from my fork to yours, etc, so in the end I opted to just push to your branch 😞 Feel free to revert if this does not conform to your requirements.

File: src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarArtists.tsx [hunk 1]
```diff
@@ -94,13 +94,17 @@ export class ArtworkSidebarArtists extends React.Component<ArtistsProps> {
...
+      @argumentDefinitions(
+        showFollowSuggestions: { type: "Boolean", defaultValue: true }
+      ) {
       artists {
         __id
         id
         name
         href
         ...FollowArtistButton_artist
+          @arguments(showFollowSuggestions: $showFollowSuggestions)
```
Comment (alloy) [inline src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarArtists.tsx]: Would have been even nicer if we could just do `@arguments(showFollowSuggestions: true)`, but [currently](https://github.com/facebook/relay/pull/2501) the compiler insists on only using variables.

File: src/Apps/Artwork/ArtworkApp.tsx [hunk 2]
```diff
@@ -147,13 +147,17 @@ export const ArtworkApp: React.SFC<Props> = props => {
 export const ArtworkAppFragmentContainer = createFragmentContainer(
   ArtworkApp,
   graphql`
-    fragment ArtworkApp_artwork on Artwork {
+    fragment ArtworkApp_artwork on Artwork
+      @argumentDefinitions(
```
[the remark this answers is not in the document]
Comment (alloy) [reply src/Apps/Artwork/ArtworkApp.tsx]: Yeah indeed, you _should_ be able to access global variables (which in reality all GraphQL variables are), but I don’t think you really would want that. The nice thing about the ‘local’ scope variables that Relay allows for is that you are _not_ required to define the variable in each operation that _somewhere_ in the tree would be using your fragment.

**Pull request merged**